test/integration/pod_node_constraints_test.go

```diff
@@ -0,0 +1,117 @@
+	pod := &kapi.Pod{}
+	pod.Name = "testpod"
+	pod.Spec.RestartPolicy = kapi.RestartPolicyNever
+	if len(nodeName) > 0 {
+		pod.Spec.NodeName = nodeName
+	}
+	if len(*nodeSelector) > 0 {
+		pod.Spec.NodeSelector = *nodeSelector
+	}
+	pod.Spec.Containers = []kapi.Container{
+		{
+			Name:  "container",
+			Image: "test/image",
+		},
+	}
+	return pod
+}
+
+func testPodNodeConstraintsExpectedError(errorMsg string) error {
+	attrs := admission.NewAttributesRecord(testPodNodeConstraintsPod("", &map[string]string{}), kapi.Kind("Pod"), "default", "test", kapi.Resource("pods"), "", admission.Create, nil)
+	return admission.NewForbidden(attrs, fmt.Errorf(errorMsg))
+}
+
+func TestPodNodeConstraintsAdmissionPluginDefaults(t *testing.T) {
+	config := &pluginapi.PodNodeConstraintsConfig{}
+	kclient := setupPodNodeConstraintsTest(t, config)
+	nodeSelector := &map[string]string{"bogus": "frank"}
+	_, err := kclient.Pods(testutil.Namespace()).Create(testPodNodeConstraintsPod("nodename.example.com", nodeSelector))
+	if err != nil {
+		t.Fatalf("Unexpected: %v", err)
+	}
+}
+
+func TestPodNodeConstraintsAdmissionPluginProhibitNodeName(t *testing.T) {
+	config := &pluginapi.PodNodeConstraintsConfig{
+		ProhibitNodeName: true,
+	}
+	kclient := setupPodNodeConstraintsTest(t, config)
+	expectedError := testPodNodeConstraintsExpectedError("use of NodeName field in pod spec prohibited by policy")
+	nodeSelector := &map[string]string{"bogus": "frank"}
+	_, err := kclient.Pods(testutil.Namespace()).Create(testPodNodeConstraintsPod("nodename.example.com", nodeSelector))
+	if err == nil {
+		t.Fatalf("Expected error %q, no error received", expectedError.Error())
+	}
+	if err.Error() != expectedError.Error() {
+		t.Errorf("expected error %q, got: %q", expectedError.Error(), err.Error())
+	}
+}
+
+func TestPodNodeConstraintsAdmissionPluginProhibitNodeSelector(t *testing.T) {
+	config := &pluginapi.PodNodeConstraintsConfig{
+		ProhibitNodeSelector: true,
+	}
+	kclient := setupPodNodeConstraintsTest(t, config)
+	expectedError := testPodNodeConstraintsExpectedError("use of NodeSelector field in pod spec prohibited by policy")
+	nodeSelector := &map[string]string{"bogus": "frank"}
+	_, err := kclient.Pods(testutil.Namespace()).Create(testPodNodeConstraintsPod("nodename.example.com", nodeSelector))
+	if err == nil {
+		t.Fatalf("Expected error %q, no error received", expectedError.Error())
+	}
+	if err.Error() != expectedError.Error() {
+		t.Errorf("expected error %q, got: %q", expectedError.Error(), err.Error())
+	}
+}
+
+func setupPodNodeConstraintsTest(t *testing.T, pluginConfig *pluginapi.PodNodeConstraintsConfig) kclient.Interface {
+	masterConfig, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("error creating config: %v", err)
+	}
+	plugins := append([]string{"PodNodeConstraints"}, kubemaster.AdmissionPlugins...)
+	masterConfig.KubernetesMasterConfig.AdmissionConfig.PluginOrderOverride = plugins
```

Comment:
You don't need to set this. Your plugin is already part of the default chain, so no need to override.